vllm/worker/model_runner.py

```diff
@@ -527,7 +526,7 @@ def prepare_input_tensors(
                 "lora_requests": lora_requests,
                 "lora_mapping": lora_mapping,
             }
-            metadata_dict.update(dataclasses.asdict(input_metadata))
+            metadata_dict.update(input_metadata.asdict_zerocopy())
```

Comment:
can you just do `input_metadata.__dict__` to avoid the new method at all?

Reply:
Perhaps, though that may lead to unexpected interactions if `InputMetadata` has extra properties added and such.